### PR TITLE
Update jenkinsExtractor.groovy

### PR DESCRIPTION
**Pull request by carmosin**
The current version doesn't support jobs with pathlike names.

The change in this pull request aims to solve this.

### CODE REVIEW

File: vars/jenkinsExtractor.groovy [hunk 1]
```diff
@@ -13,16 +15,14 @@ def call(def jobNames, def numberOfHoursBack) {
...
 	}
-    	for (def jobName: jobNames) {
-		echo "Looking for the job with the name $jobName"
-		def job = Jenkins.instance.getItem(jobName)
-  		if (job == null) {
-          		echo "Job {jobName} wasn't found. Check your configuration"
-			continue
-        	}
-		def builds = job.getBuilds().byTimestamp(System.currentTimeMillis()-numberOfHoursBack*60*60*1000, System.currentTimeMillis()).completedOnly()
+    	jobNames.each {
```
Comment (Andrey9kin) [inline vars/jenkinsExtractor.groovy]: jobNames supposed to be a list of strings - now you made it to list of objects. When user will pass list of strings this will fail since string doesn't have .fullName field

File: vars/jenkinsExtractor.groovy [hunk 2]
```diff
@@ -13,16 +15,14 @@ def call(def jobNames, def numberOfHoursBack) {
 	def buildResults = []
 	if (jobNames==[]) {
 		echo "No job names specified for extraction. Check all jobs"
-		jobNames = Jenkins.instance.getJobNames()
+		jobNames = Jenkins.instance.getAllItems(Job.class)
 	}
-    	for (def jobName: jobNames) {
-		echo "Looking for the job with the name $jobName"
-		def job = Jenkins.instance.getItem(jobName)
-  		if (job == null) {
```
Comment (Andrey9kin) [inline vars/jenkinsExtractor.groovy]: What happen to this check? Why was it removed?

File: vars/jenkinsExtractor.groovy (hunk 1 above)
Comment (Andrey9kin) [inline vars/jenkinsExtractor.groovy]: Why did you change for loop to each? Was it broken? I mean why to change something that works and not related to the issue in hand?

File: vars/jenkinsExtractor.groovy [hunk 3]
```diff
@@ -13,16 +15,14 @@ def call(def jobNames, def numberOfHoursBack) {
...
-          		echo "Job {jobName} wasn't found. Check your configuration"
-			continue
-        	}
-		def builds = job.getBuilds().byTimestamp(System.currentTimeMillis()-numberOfHoursBack*60*60*1000, System.currentTimeMillis()).completedOnly()
+    	jobNames.each { 
+		def job = it
+		def jobName = job.fullName
+		echo "Looking for the job with the name ${jobName}"
+		
+		def builds = job.builds.byTimestamp(System.currentTimeMillis()-numberOfHoursBack*60*60*1000, System.currentTimeMillis()).completedOnly()
```
Comment (Andrey9kin) [inline vars/jenkinsExtractor.groovy]: Please keep getBuilds() - internal field name might change but function supposed to return correct result even if field was renamed

File: vars/jenkinsExtractor.groovy (hunk 2 above)
Comment (carmosin) [inline vars/jenkinsExtractor.groovy]: It figured it will never go into this clause since basing the loop on objects instead of getting each item inside the loop, and didn't see why it should be kept.

**Comment by carmosin** [comment]
This change is not needed.
